### DISCUSSION TITLE
How to deploy to Docker ?

### DISCUSSION THREAD

**Discussion by chrisarts** [nandorojo/solito#61] [General]
I try to follow instructions from nextJS docs but it does not found the node_modules from packages/app so I tries to do a local build with minimun steps to copy standalone and public folder into docker and after that run the serve.js but it fails to load dependencies.

Here mi docker file:

```
FROM node:16-alpine
WORKDIR /app

ENV NODE_ENV production
# Uncomment the following line in case you want to disable telemetry during runtime.
# ENV NEXT_TELEMETRY_DISABLED 1

RUN addgroup --system --gid 1001 nodejs
RUN adduser --system --uid 1001 nextjs

# You only need to copy next.config.js if you are NOT using the default configuration
COPY  next.config.js ./
COPY public ./public
COPY package.json ./package.json

# Automatically leverage output traces to reduce image size 
# https://nextjs.org/docs/advanced-features/output-file-tracing
COPY .next/standalone ./
COPY .next/static ./.next/static

USER nextjs

EXPOSE 3000

ENV PORT 3000

CMD ["node", "server.js"]

```

(Im a Mobile developer so i dont have much experience with web envs)

**Comment by nandorojo** [comment]
Hi, a few things:

1. why are you using docker for this? i think it’s too complicated, you should just use Vercel
2. have you tried using the starter app?

**Comment by chrisarts** [comment]
Mi Infrastructure is based on DigitalOcean Kubernetes, thats why im using Docker, same behaviour with starter app.

**Comment by nandorojo** [comment]
this isn't something I'll be able to help with. I also don't exactly understand why a Next.js app would need such a complicated setup instead of just using Vercel, but I'll move this to a discussion

**Reply by chrisarts** [reply]
Actually I move my front end to vercel, but I loose my capabilities to use sockets and peerJS for WEBRTC streaming, but for the price its ok im moving my backend to a serverful service.

**Reply by nandorojo** [reply]
you can presumably keep your backend decoupled from your next.js site

**Reply by chrisarts** [reply]
yes, thats what im doing, but webrtc sessions are not supported on serverless functions unfortunately, I did a trick to keep a socket server running on the server side but  its not available to join sockets just emit data.

**Reply by chrisarts** [reply]
Also the mongoDB connection is being fired on every request, I solve it but after that for some reason the refferences does not work.

**Reply by nandorojo** [reply]
your backend doesn’t need to be related to next.js. in any case, this is beyond the scope of solito, which is only concerned with the frontend

**Reply by chrisarts** [reply]
BTW This project is awesome, solve many problems in my app, y read your document about things you preffer to know before use this approach, your advices were very helpful.

**Reply by chrisarts** [reply]
> your backend doesn’t need to be related to next.js. in any case, this is beyond the scope of solito, which is only concerned with the frontend

Yes I know that, thanks for helping me with this vercel is awesome, and really cheap for MVP's

**Comment by frontendschmacktend** [comment]
@chrisarts Were you able to fix this issue? We're facing the same issue getting the standalone version to run on Docker so we can deploy on Google Cloud Run. Unfortunately Vercel is not an option for us either otherwise we would just use that.

**Reply by nandorojo** [reply]
I wonder if the real difficulty is deploying Next.js to docker or the like?

**Reply by nandorojo** [reply]
Or maybe you need to do something special to deploy a monorepo

**Reply by frontendschmacktend** [reply]
@nandorojo The issue seems to be that the transpilePackages are not being added to the standalone build

**Reply by frontendschmacktend** [reply]
We can see the rest of the packages being added but not the ones included in the transpilePackages list

**Reply by nandorojo** [reply]
And does building locally work

**Reply by frontendschmacktend** [reply]
Yes dev build works (next dev) and a normal build (next build and next start but without the output being standalone) works

**Reply by frontendschmacktend** [reply]
This is our next.config.js if that helps:
<img width="619" alt="Screenshot 2023-12-21 at 8 55 55 AM" src="https://github.com/nandorojo/solito/assets/153449693/3350c50b-0d88-4743-9b93-d4d512c53e75">

**Reply by frontendschmacktend** [reply]
We also tried this on the starter repo yesterday and still faced the same issue

**Reply by frontendschmacktend** [reply]
If you can help us getting it to work on the starter repo we can take it from there
